ITR/data/base_providers.py

```diff
@@ -55,47 +60,57 @@ def _get_projected_production(self, scope: EScope = EScope.S1S2) -> pd.DataFrame
+    def get_benchmark_projections(self, company_sector_region_scope: pd.DataFrame) -> pd.DataFrame:
         """
         Overrides subclass method
         returns a Dataframe with production benchmarks per company_id given a region and sector.
-        :param company_sector_region_info: DataFrame with at least the following columns :
-        ColumnsConfig.COMPANY_ID, ColumnsConfig.SECTOR and ColumnsConfig.REGION
+        :param company_sector_region_scope: DataFrame indexed by ColumnsConfig.COMPANY_ID
+        with at least the following columns: ColumnsConfig.SECTOR, ColumnsConfig.REGION, and ColumnsConfig.SCOPE
         :param scope: a scope
         :return: A DataFrame with company and intensity benchmarks per calendar year per row
         """
-        benchmark_projection = self._get_projected_production(scope)  # TODO optimize performance
-        sectors = company_sector_region_info[self.column_config.SECTOR]
-        regions = company_sector_region_info[self.column_config.REGION]
-        scopes = [EScope.S1S2] * len(sectors)
-        benchmark_regions = regions.copy()
-        mask = benchmark_regions.isin(benchmark_projection.reset_index()[self.column_config.REGION])
-        benchmark_regions.loc[~mask] = "Global"
 
-        benchmark_projection = benchmark_projection.loc[list(zip(benchmark_regions, sectors, scopes))]
-        benchmark_projection.index = sectors.index
-        return benchmark_projection
+        benchmark_projection = self._get_projected_production()  # TODO optimize performance
+        df = (company_sector_region_scope[['sector', 'region', 'scope']]
+              .reset_index()
+              .drop_duplicates()
+              .set_index(['company_id', 'scope']))
+        # We drop the meaningless S1S2 from the production benchmark and replace it with the company's scope.
+        # This is needed to make indexes align when we go to multiply production times intensity for a scope.
+        company_benchmark_projections = df.merge(benchmark_projection.droplevel('scope'),
+                                                 left_on=['sector', 'region'], right_index=True, how='left')
+        mask = company_benchmark_projections.iloc[:, -1].isna()
+        if mask.any():
+            # Patch up unknown regions as "Global"
+            global_benchmark_projections = df[mask].merge(benchmark_projection.loc[(slice(None), 'Global'), :].droplevel(['region','scope']),
+                                                          left_on=['sector'], right_index=True, how='left').drop(columns='region')
+            combined_benchmark_projections = pd.concat([company_benchmark_projections[~mask].drop(columns='region'),
+                                                        global_benchmark_projections])
+            return combined_benchmark_projections.drop(columns='sector')
+        return company_benchmark_projections.drop(columns=['sector', 'region'])
```

Comment:
why do we drop sector and region?
don't we need these columns in this dataframe any more?

Reply:
I've updated the comment about the return value to read:

```
        :return: An all-quantified DataFrame with intensity benchmark data per calendar year per row, indexed by company.                                                                                                    
```
The return value of this function is later processed by cumulative sum and product functions that can only deal with dataframes that are 100% math-able.  If in the future we want to split companies into sectoral lines of business or regional entities, the company_id will have to distinguish those as a true index.

Reply:
I think that this answer, plus the documentation how of how the cumulative sum and product math works resolves this question.  Please unresolve if you disagree.